support-frontend/app/controllers/DigitalSubscription.scala

```diff
@@ -5,46 +5,51 @@ import admin.settings.{AllSettings, AllSettingsProvider, SettingsSurrogateKeySyn
 import assets.{AssetsResolver, RefPath, StyleContent}
 import cats.implicits._
 import com.gu.identity.model.{User => IdUser}
+import com.gu.monitoring.SafeLogger
+import com.gu.monitoring.SafeLogger._
 import com.gu.support.catalog.DigitalPack
-import com.gu.support.config.{PayPalConfigProvider, StripeConfigProvider}
+import com.gu.support.config.{PayPalConfigProvider, Stage, Stages, StripeConfigProvider}
+import com.gu.support.encoding.CustomCodecs._
 import com.gu.support.pricing.PriceSummaryServiceProvider
 import config.StringsConfig
-import io.circe.syntax._
-import lib.PlayImplicits._
-import com.gu.monitoring.SafeLogger
-import com.gu.monitoring.SafeLogger._
 import play.api.libs.circe.Circe
-import play.api.mvc.{request, _}
+import play.api.mvc._
 import play.twirl.api.Html
 import services._
-import views.html.subscriptionCheckout
-import views.html.helper.CSRF
-import views.ViewHelpers._
-import com.gu.support.encoding.CustomCodecs._
 import views.EmptyDiv
+import views.ViewHelpers._
+import views.html.helper.CSRF
+import views.html.subscriptionCheckout
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class DigitalSubscription(
-    priceSummaryServiceProvider: PriceSummaryServiceProvider,
-    val assets: AssetsResolver,
-    val actionRefiners: CustomActionBuilders,
-    identityService: IdentityService,
-    testUsers: TestUserService,
-    membersDataService: MembersDataService,
-    stripeConfigProvider: StripeConfigProvider,
-    payPalConfigProvider: PayPalConfigProvider,
-    components: ControllerComponents,
-    stringsConfig: StringsConfig,
-    settingsProvider: AllSettingsProvider,
-    val supportUrl: String,
-    fontLoaderBundle: Either[RefPath, StyleContent])
-(implicit val ec: ExecutionContext) extends AbstractController(components) with GeoRedirect with CanonicalLinks with Circe with SettingsSurrogateKeySyntax {
+  priceSummaryServiceProvider: PriceSummaryServiceProvider,
+  val assets: AssetsResolver,
+  val actionRefiners: CustomActionBuilders,
+  identityService: IdentityService,
+  testUsers: TestUserService,
+  membersDataService: MembersDataService,
+  stripeConfigProvider: StripeConfigProvider,
+  payPalConfigProvider: PayPalConfigProvider,
+  components: ControllerComponents,
+  stringsConfig: StringsConfig,
+  settingsProvider: AllSettingsProvider,
+  val supportUrl: String,
+  fontLoaderBundle: Either[RefPath, StyleContent],
+  stage: Stage
+)(
+  implicit val ec: ExecutionContext
+) extends AbstractController(components) with GeoRedirect with CanonicalLinks with Circe with SettingsSurrogateKeySyntax {
 
   import actionRefiners._
 
   implicit val a: AssetsResolver = assets
 
+  val stripeSetupIntentEndpoint: String =
+    if (stage == Stages.PROD) "https://stripe-intent.support.guardianapis.com/stripe-intent"
```

Comment:
This should go into the appropriate public config file eg. `PROD.public.conf` or `UAT.public.conf`